lib/perl/Genome/InstrumentData/AlignmentResult/Speedseq.pm

```diff
@@ -13,13 +13,82 @@ class Genome::InstrumentData::AlignmentResult::Speedseq {
     ],
 };
 
-sub _run_aligner {
+sub post_create {
+    return 1;
+}
+
+sub get_bam_file {
+    my $self = shift;
+
+    # Create doesn't do all of the necessary post-processing. This is being
+    # delayed until the first time that the bam file is revivified.
+    # If we don't have an allocation then this is the first revivification and
+    # we will need to do post-processing.
+    my $guard = $self->get_speedseq_bam_lock->unlock_guard();
+    unless ($self->disk_allocations) {
+        return $self->_inititalize_revivified_bam;
+    }
+    else {
+        return $self->SUPER::get_bam_file;
+    }
+}
+
+sub get_speedseq_bam_lock {
     my $self = shift;
 
-    #Run get_bam_file to fake revivify the per-lane bams
+    my $resource = File::Spec->join('genome', __PACKAGE__, 'lock-per-lane-alignment-'.$self->id);
+    my $lock = Genome::Sys::LockProxy->new(
+        resource => $resource,
+        scope => 'site',
+    )->lock(
+        max_try => 288, # Try for 48 hours every 10 minutes
+        block_sleep => 600,
+    );
+
+    die $self->error_message("Unable to acquire the lock for per lane alignment result id (%s) !", $self->id) unless $lock;
+    return $lock;
+}
+
+sub _inititalize_revivified_bam {
+    my $self = shift;
+
+    $self->_create_disk_allocation;
+
+    $self->debug_message("Prepare working directories...");
+    $self->_prepare_working_and_staging_directories;
+    $self->debug_message("Staging path is " . $self->temp_staging_directory);
+    $self->debug_message("Working path is " . $self->temp_scratch_directory);
+
     $self->_prepare_output_directory;
-    $self->get_bam_file;
-    return 1;
+
+    my $bam_file = $self->SUPER::get_bam_file;
+
+    $self->debug_message("Postprocessing & Sanity Checking BAM file (if necessary)...");
```

Comment:
Below line 66 - 89 are duplicates of lower part of create method in base class AR.pm except for commented out step description. It would be better to reduce the redundant code.